client/my-sites/marketplace/components/billing-interval-switcher/index.tsx

```diff
@@ -54,6 +54,22 @@ const BillingIntervalSwitcher: FunctionComponent< Props > = ( props: Props ) =>
 	const monthlyLabel = translate( 'Monthly' );
 	const annualLabel = translate( 'Annually' );
 
+	const searchParams = new URLSearchParams( document.location.search );
+	const billingIntervalParam = searchParams.get( 'interval' );
+
+	/**
+	 * Change the billing period based on query params, if passed
+	 */
+	useEffect( () => {
+		if ( billingIntervalParam === 'monthly' ) {
+			onChange( IntervalLength.MONTHLY );
+		}
+
+		if ( billingIntervalParam === 'annually' ) {
+			onChange( IntervalLength.ANNUALLY );
+		}
```

Comment:
Thanks @WBerredo for addressing my previous feedback. Can we use [castIntervalLength](https://github.com/Automattic/wp-calypso/blob/039a4698993fd62fee2641173a0e3876a2cc3499/client/my-sites/email/email-providers-comparison/interval-length/index.ts#L1) to checking for valid value? I believe, it will simply the logic and will help us get rid of similar code block. 

```suggestion
           const billingInterval = castIntervalLength(  billingIntervalParam  );
           if ( billingInterval  ) {
                     onChange( billingInterval );
           } 
```

Reply:
Thanks for the suggestion, but this method [castIntervalLength](https://github.com/Automattic/wp-calypso/blob/039a4698993fd62fee2641173a0e3876a2cc3499/client/my-sites/email/email-providers-comparison/interval-length/index.ts#L1) uses a different `IntervalLength` which is lowercased, which wouldn't work 1:1 as the `onChange` method expects it uppercased. 

I thought about creating a new `castIntervalLength` for the `IntervalLenght` we use, but we would need to:
* Create ifs relating the lowercased version with the uppercased version, or
* Make the comparison and then uppercase the return, but I think our current code is less complex than that.

---

Please let me know what you think about it.

Reply:
The underlying function [setBillingInterval](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/marketplace/billing-interval/actions.ts#L6) which will be eventually [called](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plugins/plugin-details-CTA/index.jsx#L338C7-L338C15) on `onChange` has expectation of either value from IntervalLength defined [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/marketplace/billing-interval/actions.ts#L7). 

So based on your suggestion, we probably need to update signature of [onChange](https://github.com/Automattic/wp-calypso/blob/3d5cab87af883a9d029ab30d6f8653333f0322a9/client/my-sites/marketplace/components/billing-interval-switcher/index.tsx#L39) which is looking wrong. 

I might be overlooking something here.

Reply:
We may need change following:
```js
onChange: ( selectedValue: 'MONTHLY' | 'ANNUALLY' ) => void;
```
With following: 
```js
onChange: ( selectedValue:  IntervalLength.MONTHLY | IntervalLength.ANNUALLY ) => void;
```

Let me know what do you think about it.

Reply:
Sorry about the whole confusion. There are two `IntervalLength` in codebase which has exactly same thing. defined [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/email/email-providers-comparison/interval-length/index.ts#L1) and [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/marketplace/components/billing-interval-switcher/constants.tsx#L1).